### PR TITLE
MatrixChainOrder

### PR DESCRIPTION
**Pull request by shivg7706**
Adding Dynamic programming Implementation of Matrix Chain order

### CODE REVIEW

**Comment by shivg7706** [comment]
@keon Is there any ambiguity in above algorithm?

**Comment by shivg7706** [comment]
@keon Is something wrong with this algorithm? Please reply.

File: dp/matrix_chain_order.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+Implementation of Matrix Chain Multiplication
+Time Complexity: O(n^3)
+Space Complexity: O(n^2)
+'''
+def MatrixChainOrder(array):
+    N=len(array)
+    Matrix=[[0 for x in range(N)] for x in range(N)]
+    Sol=[[0 for x in range(N)] for x in range(N)]
+    for i in range(1,N):
+        Matrix[i][i]=0
```
Comment (goswami-rahul) [inline dp/matrix_chain_order.py]: The loop
```python
for i in range(1,N):
       Matrix[i][i]=0
```
is redundant. 
Also the declaration of Matrix  and Sol can be reduced to
```python
Matrix = [[0] * n] * n
Sol = [[0] * n] * n
```

**Comment by goswami-rahul** [comment]
Rest code is ok.
Should be merged anytime soon

**Comment by shivg7706** [comment]
is it alright @goswami-rahul @keon ?

**Comment by goswami-rahul** [comment]
@shivg7706  Its not correct.
See my review and  match your changes.
There a bug with square brackets in both lines.

**Comment by goswami-rahul** [comment]
It is appreciated to check ( or test )  the code before requesting a PR.
Otherwise, great job done in the PR. @shivg7706

**Comment by shivg7706** [comment]
@goswami-rahul Changes suggested by you is giving an error. I couldn't able to resolve the problem in the snippets.

**Comment by goswami-rahul** [comment]
There was a bug, I think. It only created one list for the whole matrix.
This code will work just fine.
```python
Mat = [[0] * n for _ in range(n)]
```
Same for Sol matrix.
Or, you may keep it to your original version

**Pull request merged**